src/main/java/com/mongodb/kafka/connect/sink/converter/LazyBsonDocument.java

```diff
@@ -58,6 +58,8 @@ public LazyBsonDocument(
       final BiFunction<Schema, Object, BsonDocument> converter) {
     if (record == null) {
       throw new IllegalArgumentException("SinkRecord can not be null");
+    } else if (dataType == null) {
+      throw new IllegalArgumentException("dataType can not be null");
```

Comment:
The neighbour exception messages mention types of method params, not names (for example, `SinkRecord` instead of `record`). It'd be better to use a consistent scheme. I believe that param names are more useful, then param types, but regardless of the scheme you chose, consistency is of the essence.

Reply:
Yeah agreed - good spot.  I've updated it to use the param name, which follows the conventions over from the notNull method in the java driver.  This class is only used internally so users should never see these messages but for the sake of completeness they are added.